src/Security/Authentication/WsFederation/src/WsFederationHandler.cs

```diff
@@ -327,6 +369,42 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
         }
     }
 
+    private async Task<TokenValidationParameters> SetupTokenValidationParametersAsync()
+    {
+        // Clone to avoid cross request race conditions for updated configurations.
+        var tokenValidationParameters = Options.TokenValidationParameters.Clone();
+
+        if (Options.ConfigurationManager is BaseConfigurationManager baseConfigurationManager)
+        {
+            tokenValidationParameters.ConfigurationManager = baseConfigurationManager;
+        }
+        else
+        {
+            if (Options.ConfigurationManager != null)
+            {
+                // GetConfigurationAsync has a time interval that must pass before new http request will be issued.
+                _configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
```

Comment:
Why does the WsFederationHandler still cache the configuration in a field, but the JwtBearer doesn't?

See also the conversation at https://github.com/dotnet/aspnetcore/pull/48966#discussion_r1269654386.

Reply:
We can make the same change to this handler if you want. It made sense to remove the field in JwtBearer to clean up the code. It was just being read in one place.

Was trying to make less changes to WsFed where possible.

Reply:
Ok. Let’s make a follow up change in main for this.